fmriprep/viz/reports.py

```diff
@@ -147,18 +147,42 @@ def index_error_dir(self, error_dir):
                         else:
                             scope.set_extra(k, v)
                     scope.level = 'fatal'
-                    message = node_name + ': ' + gist + '\n\n' + exception_text
-
-                    # remove file paths
-                    fingerprint = re.sub(r"(/[^/ ]*)+/?", '', message)
-                    # remove words containing numbers
-                    fingerprint = re.sub(r"([a-zA-Z]*[0-9]+[a-zA-Z]*)+", '', fingerprint)
-                    # adding the return code if it exists
-                    for line in message.split('\n'):
-                        if line.startswith("Return code"):
-                            fingerprint += line
+
+                    # Group common events with pre specified fingerprints
+                    fingerprint_dict = {'permission-denied': ["PermissionError: [Errno 13] "
+                                                              "Permission denied"],
+                                        'memory-error': ["MemoryError", "Cannot allocate memory"],
+                                        'reconall-already-running': ["ERROR: it appears that "
+                                                                     "recon-all is already "
+                                                                     "running"],
+                                        'no-disk-space': ["OSError: [Errno 28] No space left on "
+                                                          "device", "[Errno 122] Disk quota "
+                                                                    "exceeded"],
+                                        'sigkill': ["Return code: 137"]}
+
+                    fingerprint = ''
+                    issue_title = node_name + ': ' + gist
+                    for new_fingerprint, error_snippets in fingerprint_dict.items():
+                        for error_snippet in error_snippets:
+                            if error_snippet in crash_info['traceback']:
```

Comment:
Are these all `str.startswith` matches? Because `in` will be a more extensive search in a potentially large traceback string. Normally wouldn't mind, but this is in a loop, and the dictionary is likely to grow, so this could start becoming a non-trivial.

Reply:
Okay. Since it's not a quick replacement, I'm okay leaving it as is.